### PR TITLE
First code piece of OS gateway, and linking to vtgate and health check

### PR DESCRIPTION
**Pull request by guoliang100**
@alainjobart @sougou

### CODE REVIEW

File: go/cmd/vtgate/vtgate.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ var (
 	connTimeoutPerConn = flag.Duration("conn-timeout-per-conn", 1500*time.Millisecond, "vttablet connection timeout (per connection)")
 	connLife           = flag.Duration("conn-life", 365*24*time.Hour, "average life of vttablet connections")
 	maxInFlight        = flag.Int("max-in-flight", 0, "maximum number of calls to allow simultaneously")
+	additionalGateway  = flag.String("additional-gateway", "", "additional gateway to enable tablet health check module")
```
Comment (alainjobart) [inline go/cmd/vtgate/vtgate.go]: All new flags should use _ as separator, not -.

File: go/vt/vtgate/discoverygateway.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	pbt "github.com/youtube/vitess/go/vt/proto/topodata"
+	tproto "github.com/youtube/vitess/go/vt/tabletserver/proto"
+	"github.com/youtube/vitess/go/vt/tabletserver/tabletconn"
+	"github.com/youtube/vitess/go/vt/topo"
+)
+
+var (
+	cellsToWatch        = flag.String("cells-to-watch", "", "comma-separated list of cells for watching endpoints")
+	refreshInterval     = flag.Duration("endpoint-refresh-interval", 1*time.Minute, "endpoint refresh interval")
+	topoReadConcurrency = flag.Int("topo-read-concurrency", 32, "concurrent topo reads")
```
Comment (alainjobart) [inline go/vt/vtgate/discoverygateway.go]: same here, _ as separator.

**Comment by alainjobart** [comment]
OK I am 100% confused, why do we need an additional gateway? I thought we were going to have only one, the one you already implemented the interface for (which is the old one), and the one based on the discovery you added earlier would be another flavor (the new one, for open source). What is this about?

File: go/cmd/vtgate/vtgate.go (hunk 1 above)
Comment (guoliang100) [inline go/cmd/vtgate/vtgate.go]: Done.

File: go/vt/vtgate/discoverygateway.go (hunk 2 above)
Comment (guoliang100) [inline go/vt/vtgate/discoverygateway.go]: Done.

**Comment by guoliang100** [comment]
I want to test the healthcheck module first before having the full implementation of new gateway.
So the additional gateway is to turn on the healthchecking even if we use existing gateway.

**Comment by alainjobart** [comment]
Ah OK, then the name is a bit confusing, maybe change it to testGateway, and add a clear comment that this is meant to test the new code.

**Comment by alainjobart** [comment]
LGTM then

**Pull request merged**